### ISSUE TITLE
Terminal title truncation on maximise/fullscreen

### ISSUE THREAD

**Issue by alexwhitman** [gnunn1/terminix#551]
I've set my terminal title to `[${columns}x${rows}] ${title}`.  If I open a terminix window and then maximise the title is truncated at the start and appears as `...x76] alex@machine: ~`.  Triggering a title redraw, such as entering a line in the terminal, fixes the title.

I can also reproduce with just `${columns}` as the title.

**Comment by gnunn1** [comment]
I can't reproduce this behavior in either Arch or Ubuntu 16.04. I tried reproducing it by splitting the terminal and switching to a long directory so the '...' ellipsis appears at the beginning. I then maximize the terminal and the title adjusts fine for the extra space and the ellipsis disappears. I also tried fullscreen and that worked fine as well.

Are there any non-default options you might have set that would be relevant to the behavior?

**Comment by alexwhitman** [comment]
I've got some settings as non-default but I've played around with them and so far nothing has changed.  I've also changed system fonts and sizes but that doesn't fix it either.

**Comment by gnunn1** [comment]
If you want to try something, you can add the following lines to terminal.d:370

```
lblTitle.setHexpand(true);
lblTitle.setHalign(Align.FILL);
```

**Comment by alexwhitman** [comment]
That just centered the button in the terminal title bar, but the truncation still occurs.

**Comment by gnunn1** [comment]
@alexwhitman Are you still experiencing this issue?

**Comment by gnunn1** [comment]
@alexwhitman Can you let me know when you get a chance if the fix I checked in for #779 takes care of this as well.

**Comment by alexwhitman** [comment]
@gnunn1 I've been out of the country for a bit but I'll check this in the next few days.

**Comment by alexwhitman** [comment]
@gnunn1 I've had a chance to test and the same issue is present.

**Comment by gnunn1** [comment]
Bummer, when you say maximize I assume you mean the window and not the terminal? I'll try adding an event to window state change and doing the queueRedraw trick I'm using in the other bug report.

**Comment by alexwhitman** [comment]
Yes, maximising the window.  I can make a video if it helps.

**Comment by gnunn1** [comment]
No need, that should be fine.

**Comment by gnunn1** [comment]
I'm still thinking about this, the issue is the state change message happens in the window but I need to do the queueResize on a very specific widget in the terminal. Just trying to think of the best way while maintaining limited terminal visibility to the window.

**Comment by gnunn1** [comment]
So I've finally been able to reproduce this in 16.04 however it works fine in 16.10. As far as I can tell, this is a GTK bug. I could workaround it but it would involve some messiness to propagate the wondow state change down to the control. Since this bug appears to be rectified in 16.10 I'm going to mark this as Won't Fix.

**Comment by alexwhitman** [comment]
That's fair, it's only a really minor thing anyway which disappears once anything else is done with the window.
